SimPEG/EM/NSEM/FieldsNSEM.py

```diff
@@ -0,0 +1,563 @@
+from SimPEG import Survey, Utils, Problem, np, sp, mkvc
+from scipy.constants import mu_0
+import numpy as np
+import scipy.sparse as sp
+from SimPEG.Utils import Zero, Identity
+from SimPEG.EM.Utils import omega
+
+
+##############
+### Fields ###
+##############
+class BaseNSEMFields(Problem.Fields):
+    """Field Storage for a NSEM method."""
+    knownFields = {}
+    dtype = complex
+
+###########
+# 1D Fields
+###########
+class Fields1D_ePrimSec(BaseNSEMFields):
+    """
+    Fields storage for the 1D NSEM solution.
+
+    Solving for e fields, using primary/secondary formulation
+    """
+    knownFields = {'e_1dSolution':'F'}
+    aliasFields = {
+                    'e_1d' : ['e_1dSolution','F','_e'],
+                    'e_1dPrimary' : ['e_1dSolution','F','_ePrimary'],
+                    'e_1dSecondary' : ['e_1dSolution','F','_eSecondary'],
+                    'b_1d' : ['e_1dSolution','E','_b'],
+                    'b_1dPrimary' : ['e_1dSolution','E','_bPrimary'],
+                    'b_1dSecondary' : ['e_1dSolution','E','_bSecondary']
+                  }
+
+    def __init__(self,mesh,survey,**kwargs):
+        BaseNSEMFields.__init__(self,mesh,survey,**kwargs)
+
+    def _ePrimary(self, eSolution, srcList):
+        """
+        Primary electric field from source
+
+        :param numpy.ndarray eSolution: field we solved for
+        :param list srcList: list of sources
+        :rtype: numpy.ndarray
+        :return: primary electric field as defined by the sources
+        """
+        ePrimary = np.zeros_like(eSolution)
+        for i, src in enumerate(srcList):
+            ep = src.ePrimary(self.survey.prob)
+            if ep is not None:
+                ePrimary[:,i] = ep[:,-1]
```

Comment:
if you start using the Zero(), it will be better to do:
```
ePrimary[:,i] = ePrimary[:,i] + ep[:,-1]
```
as ep will not be None, but might be Zero()